modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/service/impl/KaleoInstanceLocalServiceImpl.java

```diff
@@ -393,18 +394,20 @@ kaleoDefinitionName, null, _getClassNames(assetClassName), null,
 			orderByComparator, serviceContext);
 
 		return new BaseModelSearchResult<>(
-			Stream.of(
-				hits.getDocs()
-			).map(
-				document -> GetterUtil.getLong(
-					document.get(Field.ENTRY_CLASS_PK))
-			).map(
-				kaleoInstancePersistence::fetchByPrimaryKey
-			).filter(
-				Objects::nonNull
-			).collect(
-				Collectors.toList()
-			),
+			(List<KaleoInstance>)TransformUtil.transformToList(
+				hits.getDocs(),
+				document -> {
+					KaleoInstance kaleoInstance =
+						kaleoInstancePersistence.fetchByPrimaryKey(
+							GetterUtil.getLong(
+								document.get(Field.ENTRY_CLASS_PK)));
+
+					if (kaleoInstance == null) {
```

Comment:
@JanisZhang This is redundant, please simply. There are several places in this pull like this.